bentoml/service/artifacts/__init__.py

```diff
@@ -125,6 +146,16 @@ def wrapped_load(*args, **kwargs):
                         "`load` an artifact multiple times may lead to unexpected "
                         "behaviors"
                     )
+
+                # load metadata if exists
```

Comment:
Could we do the load lazily in the `metadata` property call?
```
@property
def metadata(self):
     if not self._metadata:
          # load 
     return self._metadata
```

Reply:
I would imagine that introducing two other properties to the class like `_metadata_path` and `_metadata_loaded` is adding a lot of extra complexity for just a simple YAML load, what would be the main benefit of lazy loading here?

Reply:
@jackyzha0  got it, that makes sense. I think we would just need to add a property `_load_path`, and `if self._metadata is None` can be used to check if it's loaded.

You are right there isn't much benefit besides avoiding loading this file. I would assume most of the cases the yaml file will be very small so it should no be a problem, this LGTM!